test/test_blocks.jl

```diff
@@ -0,0 +1,76 @@
+using Test, BlockArrays
+
+@testset "blocks(::BlockVector)" begin
+    vector_blocks = [[1, 2], [3, 4, 5], Int[]]
+    @test blocks(mortar(vector_blocks)) === vector_blocks
+    @test blocks(view(mortar(vector_blocks), Block(1):Block(2))) == vector_blocks[1:2]
+end
+
+@testset "blocks(::BlockMatrix)" begin
+    matrix_blocks = permutedims(reshape([
+        1ones(1, 3), 2ones(1, 2),
+        3ones(2, 3), 4ones(2, 2),
+    ], (2, 2)))
+    @test blocks(mortar(matrix_blocks)) === matrix_blocks
+    @test blocks(view(mortar(matrix_blocks), Block(1):Block(2), Block(2):Block(2))) ==
+          matrix_blocks[1:2, 2:2]
```

Comment:
Do you want special handling for `SubArray`s?  This currently already works because of the generic fallback.

Reply:
Yes because the generic fallback is not efficient. My primary reason for adding this (which got muddled by a tangential desire for nice printing) is for fast paths to convert subviews of a `BlockMatrix{T,<:Tridiagonal}` to a `BlockBandedMatrix` see https://github.com/JuliaMatrices/BlockBandedMatrices.jl/pull/73

Reply:
OK. So I copied your code into this PR and add a few tests.